pipelines/batch/src/main/java/org/openmrs/analytics/ConvertResourceFn.java

```diff
@@ -76,7 +75,16 @@ public void writeResource(HapiRowDescriptor element)
         new Meta()
             .setVersionId(element.resourceVersion())
             .setLastUpdated(simpleDateFormat.parse(element.lastUpdated()));
+
     String jsonResource = element.jsonResource();
+    // The jsonResource field will be empty in case of deleted records and are skipped when written
+    // to target parquet files/sinkDb. This is fine for initial batch as they need not be migrated,
+    // but for incremental run they need to be migrated and the original records have to be deleted
+    // from the consolidated parquet files/sinkDb.
+    // TODO https://github.com/google/fhir-data-pipes/issues/547
+    if (jsonResource == null || jsonResource.isBlank()) {
+      return;
+    }
```

Comment:
Can you also add a unit-test in `ConvertResourceFnTest` to cover this case? Eventually we need to have some DELETE scenario covered in e2e tests as well but for now let's at least cover it in unit-tests to prevent future regressions.

Reply:
Unit test case added for the DELETE scenario